server/bleep/src/analytics.rs

```diff
@@ -57,52 +66,82 @@ pub struct HubOptions {
     pub package_metadata: Option<PackageMetadata>,
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct ApplicationSeed(String);
+
+/// User-specific configuration
+#[derive(Serialize, Deserialize)]
+pub struct UserState {
+    #[serde(default)]
+    tracking_id: String,
+}
+
 impl RudderHub {
-    pub fn new(key: String, data_plane: String) -> Arc<Self> {
-        let client = RudderAnalytics::load(key, data_plane);
-        let hub = Self {
-            client,
-            options: None,
-        };
-        let _ = HUB.set(Arc::new(hub));
-        RudderHub::get().unwrap()
+    pub fn new(
+        state: &StateSource,
+        tracking_seed: impl Into<Option<String>>,
+        key: String,
+        data_plane: String,
+    ) -> anyhow::Result<Arc<Self>> {
+        Self::new_with_options(state, tracking_seed, key, data_plane, None)
     }
 
-    pub fn new_with_options(key: String, data_plane: String, options: HubOptions) -> Arc<Self> {
+    pub fn new_with_options(
+        state: &StateSource,
+        tracking_seed: impl Into<Option<String>>,
+        key: String,
+        data_plane: String,
+        options: impl Into<Option<HubOptions>>,
+    ) -> anyhow::Result<Arc<Self>> {
         let client = RudderAnalytics::load(key, data_plane);
-        let hub = Self {
+        Ok(Self {
             client,
-            options: Some(options),
-        };
-        let _ = HUB.set(Arc::new(hub));
-        RudderHub::get().unwrap()
+            options: options.into(),
+            user_store: state.load_or_default("user_tracking")?,
+            tracking_seed: state.load_state_or("application_seed", tracking_seed.into())?,
```

Comment:
Bit unclear on the semantics here. What's the difference between:
- `tracking_seed`
- `application_seed`
- `tracking_id`